### ISSUE TITLE
Cannot exclude mathlive fields from tab keyboard navigation (tabindex=-1)

### ISSUE THREAD

**Issue by cyremur** [arnog/mathlive#544]
## Description

Mathlive textfields appear to disregard the html `tabindex` attribute.

Attempts to exclude certain Mathlive textfields from tabbing navigation with `tabindex=-1` fail.

![image](https://user-images.githubusercontent.com/5361181/87466402-7869c800-c616-11ea-9448-faeef2175127.png)

### Actual behavior

When setting `tabindex=-1` or `tabindex=100`, the mathlive textfield remains in the same place of the tab navigation.

### Expected Behavior

When setting `tabindex=-1`, the mathlive textfield is skipped in tab navigation.

## Steps to Reproduce

1. Go to https://mathlive.io/
2. Open Developer Tools
3. Click into the mathlive textfield
4. Tab and Shift+Tab to make sure you're on the element focussed by tabbing
5. Type `document.activeElement` in the Development Console to get the node focussed by tabbing
6. Set `tabindex="-1"` for that html element.
7. Repeat 3. and 4. to see if the mathlive textfield can still be focussed by tabbing

To verify that this process works correctly for other html elements (e.g. buttons), you can try the same procedure for the `Show code section` interface element in the sidebar.

1. Go to https://mathlive.io/
2. Open Developer Tools
3. Click on background
4. Use Shift+Tab
5. Type `document.activeElement` in the Development Console to get the node focussed by tabbing (should be <button id="code-section-button">)
6. Set `tabindex="-1"` for that html element.
7. Repeating 3. 4. and 5. should now show you that the `Switch Theme` element is focussed, hence `Show code section` button is correctly skipped

### Environment

**Operating System**
Windows

**Browser** 
Chrome

**URL** 
https://mathlive.io/

**Comment by Khazuar** [comment]
Can confirm this. Why are mathfields behaving so differently from other input fields?

**Comment by arnog** [comment]
That's probably a bug. Keep in mind that mathfields are not real input fields. There's a lot of work involved in being able to pretend they are like other fields. In particular, there is actually a hidden textarea associated with each mathfield that captures the keyboard input. I suspect that this is interfering with the tabbing logic. If someone wants to investigate this further and submit a fix, I'd be happy to integrate it.

**Comment by NSoiffer** [comment]
Could it have anything to do with 'onTabOutOf'?

**Comment by stefnotch** [comment]
I checked this issue out and everything seems to work as expected. (Firefox 80) The thing is, on `mathlive.io`, the mathfield itself has a tabindex.

![image](https://user-images.githubusercontent.com/10220080/93005056-92a81f80-f54d-11ea-9d25-ba29100f6ff3.png)

To verify that this is indeed what's causing, after executing the following script, you cannot tab into the mathfield anymore.

```
document.querySelector("#math-editor").setAttribute("tabindex", "-1")
[...document.querySelectorAll(".ML__textarea__textarea")].forEach(v => v.setAttribute("tabindex", "-1"))
```

**Comment by arnog** [comment]
OK, thanks, that's helpful. There is also a tabindex on the `textarea` element. That might interfere with the desired behavior (i.e. that setting the `tabindex` of the top level mathfield only should be sufficient to control its tabbing behavior.

**Comment by stefnotch** [comment]
I suppose some solutions would be to 
- copy the tabindex attribute when creating the mathfield
- use a custom element for the mathfield and use the [`attributeChangedCallback`](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks)
- use a mutation observer, which also works in Internet Explorer

**Comment by arnog** [comment]
I think using a custom element is the simplest and more robust solution.

**Comment by stefnotch** [comment]
That would break internet explorer compatibility, but I think that's fine at this point. (Even Microsoft is starting to drop support for it.)
Another thing that could be done when breaking IE compatibility is to stop shipping the `.woff` fonts and only ship the more modern `.woff2` fonts.

**Comment by arnog** [comment]
This is now working correctly when using web component, i.e. the `<math-field>` tag.
